Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx

```diff
@@ -163,6 +167,32 @@ CastImageFilter<TInputImage, TOutputImage>::DynamicThreadedGenerateDataDispatche
     inputIt.NextLine();
     outputIt.NextLine();
   }
+#else
+
+  auto inputRange = ImageRegionRange<const TInputImage>(*inputPtr, inputRegionForThread);
+  auto outputRange = ImageRegionRange<TOutputImage>(*outputPtr, outputRegionForThread);
+
+  auto       inputIt = inputRange.begin();
+  auto       outputIt = outputRange.begin();
+  const auto inputEnd = inputRange.end();
+
+
+  OutputPixelType    outputPixel{ *outputIt };
```

Comment:
At this point, the value of `*outputIt` may not yet be initialized, right? The output pixel values may still be uninitialized at this point. If so, better just declare `OutputPixelType outputPixel;` without retrieving the value from `*outputIt`.

Reply:
This is also the "BUG" cast where the reference to the first pixel is maintained and will be assigned all values in the range.

Reply:
Are you implying that #5661 didn't fix a bug? To be honest, the time it took me to understand the origin of this bug is a clear indication to me that this is not the expected behavior of the Get function of an iterator. And I guess this is not consistent with how it behaves with scalar pixel.